src/bootloaders/syslinux-common.c

```diff
@@ -23,6 +27,42 @@
 #include "system_stub.h"
 #include "writer.h"
 
+char *syslinux_common_get_default_kernel(const BootManager *manager)
+{
+        autofree(char) *config_path = NULL;
+        autofree(FILE) *f = NULL;
+        char *result = NULL;
+        char *buf = NULL;
+        size_t len, lplen = 0;
+        ssize_t read;
+        char *lookup = "DEFAULT ";
+        struct SyslinuxContext *ctx = NULL;
+
+        ctx = boot_manager_get_data((BootManager *)manager);
+
+        lplen = strlen(lookup);
+        config_path = string_printf("%s/syslinux.cfg", ctx->base_path);
+
+        f = fopen(config_path, "r");
+        CHECK_ERR_RET_VAL(!f, NULL, "Could not open config file: %s", config_path);
+
+        while ((read = getline(&buf, &len, f)) != -1) {
+                char *tk;
+
+                tk = strstr(buf, lookup);
+                if (tk != NULL) {
+                        result = strndup(tk+lplen, strlen(tk)-(lplen+1));
```

Comment:
If the DEFAULT line doesn't have a newline I think this loses a character. Maybe not a problem though in practice but I think it is safer to just use len from getline and do `result = strdup(tk+lplen, len - lplen)` and then check if the second to last character of result is a newline and convert it into a null if it is. Thoughts?

Reply:
should work, can't think why not.

Reply:
done. thanks.